core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -389,15 +390,39 @@ public static Wallet fromWatchingKey(NetworkParameters params, DeterministicKey
         return new Wallet(params, KeyChainGroup.builder(params).addChain(chain).build());
     }
 
+     /**
+     * Creates a wallet that tracks payments to and from the HD key hierarchy rooted by the given watching key. The
+     * account path is specified.
+     * @param params The network
+     * @param watchKeyB58 The key in base58 notation
+     * @param creationTime creation time of the key (if unknown use {@link #fromWatchingKeyB58(NetworkParameters, String)}
+     * @return a new wallet
+     */
+    public static Wallet fromWatchingKeyB58(NetworkParameters params, String watchKeyB58, Instant creationTime) {
+        final DeterministicKey watchKey = DeterministicKey.deserializeB58(null, watchKeyB58, params.network());
+        watchKey.setCreationTimeSeconds(creationTime.getEpochSecond());
+        return fromWatchingKey(params, watchKey, outputScriptTypeFromB58(params, watchKeyB58));
+    }
+
     /**
      * Creates a wallet that tracks payments to and from the HD key hierarchy rooted by the given watching key. The
-     * account path is specified. The key is specified in base58 notation and the creation time of the key. If you don't
-     * know the creation time, you can pass {@link DeterministicHierarchy#BIP32_STANDARDISATION_TIME_SECS}.
+     * account path is specified. They key's creation time will be set to {@link DeterministicHierarchy#BIP32_STANDARDISATION_TIME_SECS}
+     * @param params The network
+     * @param watchKeyB58 The key in base58 notation
+     * @return a new wallet
      */
+    public static Wallet fromWatchingKeyB58(NetworkParameters params, String watchKeyB58) {
+        return fromWatchingKeyB58(params, watchKeyB58, Instant.ofEpochSecond(DeterministicHierarchy.BIP32_STANDARDISATION_TIME_SECS));
+    }
+
+    /**
+     * @deprecated Use {@link #fromWatchingKeyB58(NetworkParameters, String, Instant)} or {@link #fromWatchingKeyB58(NetworkParameters, String)}
+     */
+    @Deprecated
     public static Wallet fromWatchingKeyB58(NetworkParameters params, String watchKeyB58, long creationTimeSeconds) {
-        final DeterministicKey watchKey = DeterministicKey.deserializeB58(null, watchKeyB58, params);
-        watchKey.setCreationTimeSeconds(creationTimeSeconds);
-        return fromWatchingKey(params, watchKey, outputScriptTypeFromB58(params, watchKeyB58));
+        return (creationTimeSeconds == 0)
```

Comment:
I generally like an if-else statement more for cases like this, also because if there were no return value you could not use the terniary op.

But if course this is only bikeshedding.

Reply:
I've become a much bigger fan of the ternary operator as I've been using functional-style Java. Because it is an expression and not a statement.  (I'm not a huge fan of the `boolean ? expr1 : expr2` syntax, but I try to make do and once I saw the current formatting that I generally use I found that fairly readable)

In this case I think it makes the code more reasonable because you see that all the method does it return a value.  (i.e. the whole method is simply `return expression`)

In the case of a void method (i.e. one with side-effects) it is by definition imperative (not functional) code and can not be a single expression.

I'm happy to rewrite this if you insist, though.  (Although long-run it doesn't matter because this method is deprecated for removal)

Reply:
I trust you on what's best for functional style.